### ISSUE TITLE
import/no-unresolved doesnt respect package exports

### ISSUE THREAD

**Issue by thepassle** [import-js/eslint-plugin-import#2389]
If I import something from package `foo`, that has in it's `package.json` the following exports map:

```json
"exports": {
  "./bar.js": "./src/bar.js"
}
```

The following import will cause an `import/no-unresolved` error, while being completely valid:

```js
import 'foo/bar.js';
```

**Comment by ljharb** [comment]
Duplicate of #1810.

**Comment by ljharb** [comment]
I'd suggest making that package have an actual `./bar.js` file, so that pre-exports nodes (and tooling) can locate the file also.

**Comment by thepassle** [comment]
A better structural fix would be for tooling to catch up with package exports instead

**Comment by ljharb** [comment]
Both fixes are needed; packages should remain backwards-compatible, instead of being hostile to those users.

Certainly this package, and the ecosystem's tooling, will eventually catch up as well.

**Comment by thepassle** [comment]
> packages should remain backwards-compatible, instead of being hostile to those users

The `foo` package I mentioned doesnt have this requirement

**Comment by ljharb** [comment]
I gathered that - I'm saying that all packages _should_ have that requirement.

**Comment by thepassle** [comment]
I disagree with that, but I guess its besides the point. You mentioned that "eventually" this plugin will support package exports, the issue you linked has been open for 2 years, is there already work started to support package exports? If not, would you be willing to take a PR? I might be able to spend some time and take a stab at it

**Comment by ljharb** [comment]
The work needs to be done in the `resolve` package; it has begun and is moving slowly.
